src/CofreeBot/Bot.hs

```diff
@@ -145,6 +150,35 @@ fixBot (Bot b) = go
     go :: s -> Behavior m i o
     go s = Behavior $ \i -> second go <$> b s i
 
+-- TODO: A bi-parser typeclass
+type Serializable a = (Read a, Show a)
+
+fixBotPersistent :: forall m s i o. (MonadIO m, Serializable s) => Bot m s i o -> s -> IO (Behavior m i o)
+fixBotPersistent (Bot bot) initialState = do
+  saveState "state" initialState
```

Comment:
Probably callers of this function should give a key, otherwise they will all clobber one another, right?

Reply:
Are you imagining having multiple bots running on the same linux user?

Reply:
I'm imaging each sub bot wants to manage its own state separately from the global composition

Reply:
Do you mean that you would want bots to be able to state explicitly how their individual state is persisted?

Reply:
> I'm imaging each sub bot wants to manage its own state separately from the global composition

I think @isovector is talking about a scenario in which a user  `fixBotPersistent`s several independent bots, then laterally composes the resulting `Behavior`s (or whatever it is we're calling them now), instead of doing it the other way around as we're doing now. This is a totally reasonable use case (although not one we need right now), and it'd be pretty easy to support by just adding a parameter for the file path here.

Reply:
Oh ok that is a totally reasonable user expectation. I'll thread in the path from main rather then hard code it here.

Reply:
done!